test/tests/ComputationCoordinatorSmokeTest.java

```diff
@@ -1,42 +1,47 @@
 package tests;
 
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
-
-import java.io.IOException;
-
 import src.ComputationCoordinatorImp;
 import src.ComputeEngineImp;
 import src.DataStorageImp;
-import src.MatrixImplementation;
 import src.ComputeRequest;
-import src.ComputeResult;
-
-public class ComputationCoordinatorSmokeTest {
-
-    private ComputationCoordinatorImp computationCoordinator;
-    private DataStorageImp dataStore;
-    private ComputeEngineImp computeEngine;
-    private MatrixImplementation matrixCompute;
 
-    public void setUp() {
-        dataStore = mock(DataStorageImp.class);
-        computeEngine = mock(ComputeEngineImp.class);
-        computationCoordinator = new ComputationCoordinatorImp(dataStore, computeEngine);
-    }
+import java.io.IOException;
 
-    //@Test
-    public void testBeginComputationWithRequest() throws IOException {
-        ComputeRequest request = new ComputeRequest();
-        computationCoordinator.beginComputation(request);
+/**
+ * Unit test for the ComputationCoordinatorImp class.
+ */
+public class ComputationCoordinatorSmokeTest {
 
-        assertNotNull(request);
-        // verify(computeEngine).initializeAction(anyString(), anyString(), anyString());
-        // verify(dataStore).readInputFile();
-    }
+  private ComputationCoordinatorImp computationCoordinator;
+  private DataStorageImp dataStore;
+  private ComputeEngineImp computeEngine;
+
+  /**
+   * Sets up the mock objects before each test.
+   */
+  @BeforeEach
+  public void setUp() {
+    dataStore = mock(DataStorageImp.class);
+    computeEngine = mock(ComputeEngineImp.class);
+    computationCoordinator = new ComputationCoordinatorImp(dataStore, computeEngine);
+  }
+
+  /**
+   * Tests the beginComputation method with a ComputeRequest.
+   * 
+   * @throws IOException if an input/output error occurs.
+   */
+  //@Test
+  public void testBeginComputationWithRequest() throws IOException {
```

Comment:
I don't believe you need to add a throws IOException to a Junit test. We should know if it fails using the Junit framework. 
Or this is to specifically  test our other exceptions if an invalid input is passed?

Reply:
when computerequest is made it makes a input and output class which have Exception handling so I dont know if we need it.